main.tf

```diff
@@ -73,23 +73,23 @@ resource "aws_ssm_parameter" "certificate_pem" {
   name   = "/${var.name}/iot/certificate-pem"
   type   = "SecureString"
   value  = aws_iot_certificate.default.certificate_pem
-  key_id = module.edgedevice_kms_key.id
+  key_id = var.kms_key_id != null ? var.kms_key_id : module.edgedevice_kms_key.id
   tags   = var.tags
 }
 
 resource "aws_ssm_parameter" "public_key" {
   name   = "/${var.name}/iot/public-key"
   type   = "SecureString"
   value  = aws_iot_certificate.default.public_key
-  key_id = module.edgedevice_kms_key.id
+  key_id = var.kms_key_id != null ? var.kms_key_id : module.edgedevice_kms_key.id
```

Comment:
So do we need to make the module `edgedevice_kms_key` optional as well? Or is it needed for something else?

Reply:
Needed as well (for now), if this would be removed TF will break since it cannot read the existing SSM values